### PR TITLE
Feature/virgin

### PR DESCRIPTION
**Pull request by Cardamced**
## Description

Please include a summary of the change or which issue is fixed.

## Type of change

- [X] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [X] Frontend
- [X] Backend
- [ ] Other: .....

## General checklist:

- [X] My code follows the style guidelines of this project
- [X] I ran `bun run format`/`bun run lint` before commit
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation where needed
- [X] My changes generate no new warnings
- [X] I have checked my code and corrected any misspellings
- [X] I have updated `develop` and merged to my branch before submitting pull request
- [X] My pull request generate no conflicts with `develop` branch
- [X] I requested code review from other team members

## Frontend checklist:

- [X] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [X] My `Javascript` generate no new console errors
- [X] I tested my code cross browsers
- [X] My slice is pixel perfect for both desktop and mobile according to design
- [ ] I conducted basic QA to assure all features are working
- [X] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [ ] I tested admin by manually adding content from zero
- [ ] I created easy to use admin experience which is self-explanatory
- [ ] I added description to admin fields in hard-to-understand areas
- [X] I followed guidelines for naming `Typescript` variables
- [ ] I conducted basic QA to assure all features are working

### CODE REVIEW

**Review by No3ml3 — DISMISSED** [review]

**Review by jaalazard — DISMISSED** [review]

File: packages/frontend/web/src/components/CheckBirthdate.tsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import { type FormEvent, useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 
-import CheckBirthdateAnimations from '@/components/CheckBirthdateAnimations';
-import { useAnimations } from '@/contexts/AnimationsContext';
-import { useBirth } from '@/contexts/BirthContext';
+import CheckBirthdateAnimations from '../components/CheckBirthdateAnimations';
+import { useAnimations } from '../contexts/AnimationsContext';
+import { useBirth } from '../contexts/BirthContext';
```
Comment (YDrogen) [inline packages/frontend/web/src/components/CheckBirthdate.tsx]: Pourquoi enlever les @ ?

File: packages/frontend/web/src/components/CheckBirthdateAnimations.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
-import { useAnimations } from '@/contexts/AnimationsContext';
-import { useBirth } from '@/contexts/BirthContext';
+import { useAnimations } from '../contexts/AnimationsContext';
+import { useBirth } from '../contexts/BirthContext';
```
Comment (YDrogen) [inline packages/frontend/web/src/components/CheckBirthdateAnimations.tsx]: whyyy

File: packages/frontend/web/src/components/Layout/Layout.tsx [hunk 3]
```diff
@@ -1,8 +1,7 @@
 // import { useEffect, useState } from 'react';
 import { Outlet } from 'react-router-dom';
 
-import { useAuth } from '@/contexts/AuthContext';
-
+import { useAuth } from '../../contexts/AuthContext';
```
Comment (YDrogen) [inline packages/frontend/web/src/components/Layout/Layout.tsx]: whyyyyyyyyy

File: packages/frontend/web/src/router.tsx
Comment (YDrogen) [inline packages/frontend/web/src/router.tsx:1]: hein
y'a un problème là

File: packages/frontend/web/src/components/cocktail-detail/CocktailCard.tsx
Comment (YDrogen) [inline packages/frontend/web/src/components/cocktail-detail/CocktailCard.tsx:1]: Les cocktails avec alcools n'existent plus ?

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/frontend/web/src/components/CheckBirthdate.tsx (hunk 1 above)
Comment (Cardamced) [reply packages/frontend/web/src/components/CheckBirthdate.tsx]: Parce que ça générait des erreurs... enfin... ça soulignait en rouge.

File: packages/frontend/web/src/router.tsx
Comment (Cardamced) [reply packages/frontend/web/src/router.tsx:1]: Indeed... réglage de conflits foireux. Inadvertance toussa... je me couvre la tête de cendres froides (et les relecteurs aussi).

File: packages/frontend/web/src/components/Layout/Layout.tsx (hunk 3 above)
Comment (Cardamced) [reply packages/frontend/web/src/components/Layout/Layout.tsx]: Idem... ça générait des erreurs. Mais peut-être mon VScode qui voyait la Vierge... ça m'a étonné aussi (les erreurs, pas la Vierge).

File: packages/frontend/web/src/components/CheckBirthdateAnimations.tsx (hunk 2 above)
Comment (Cardamced) [reply packages/frontend/web/src/components/CheckBirthdateAnimations.tsx]: Passkeu.

File: packages/frontend/web/src/components/cocktail-detail/CocktailCard.tsx
Comment (Cardamced) [reply packages/frontend/web/src/components/cocktail-detail/CocktailCard.tsx:1]: Vu avec Justine qui apparemment a fait des modifications. D'où la card plus nécessaire à cet endroit.

**Review by No3ml3 — DISMISSED** [review]

**Review by jaalazard — DISMISSED** [review]

File: packages/frontend/web/src/components/CheckBirthdate.tsx
Comment (YDrogen) [inline packages/frontend/web/src/components/CheckBirthdate.tsx:1]: Pourquoi toucher à ce component ? Ce n'est pas la PR pour

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/frontend/web/src/components/CheckBirthdate.tsx
Comment (Cardamced) [reply packages/frontend/web/src/components/CheckBirthdate.tsx:1]: Je suis bien d'accord et c'est ce que je commente ici ligne 22 : 
  // UseEffect commenté pour ne pas empêcher le scroll sur les pages avant le fix sur CheckBirthdate.
  
  Sinon ça empêche le scroll... Je dois faire le fix sur la PR qu'on a discuté --> régler Checkbirthdate et isUnderAge.

Comment (YDrogen) [reply packages/frontend/web/src/components/CheckBirthdate.tsx:1]: C'est donc à régler dans une autre PR, pas celle-ci 🤓

**Review by No3ml3 — DISMISSED** [review]

**Review by jaalazard — DISMISSED** [review]

**Review by No3ml3 — DISMISSED** [review]

**Review by jaalazard — DISMISSED** [review]

File: packages/frontend/web/src/pages/Virgin.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        setVirginCocktails(data.virginCocktails);
+      } catch (error) {
+        console.error('Failed to retrieve any coktail:', error);
+      }
+    };
+
+    fetchCocktails().catch((error) => {
+      error;
+    });
+  }, []);
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/Virgin.tsx]: il manque la fonction cleanup

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/frontend/web/src/pages/Virgin.tsx (hunk 4 above)
Comment (Cardamced) [reply packages/frontend/web/src/pages/Virgin.tsx]: Corrigé.

**Review by jaalazard — APPROVED** [review]

**Review by Justicesvr — APPROVED** [review]

**Review by YDrogen — APPROVED** [review]

**Pull request merged**